packages/node/src/methods/app-instance/propose-install/controller.ts

```diff
@@ -38,6 +38,11 @@ export default class ProposeInstallController extends NodeController {
     params: Node.ProposeInstallParams
   ): Promise<Queue[]> {
     const { store, publicIdentifier } = requestHandler;
+    const { initialState } = params;
+
+    if (!initialState) {
+      throw new Error(NULL_INITIAL_STATE_FOR_PROPOSAL);
+    }
```

Comment:
This belongs in the `beforeExecution` function.

Reply:
Reason for moving it here is so that the request shouldn't even get queued if it would end up throwing anyway.